### PR TITLE
Bug 1220978 - Migrate bluetooth app to notificationHelper

### PR DESCRIPTION
**Pull request by hkasemir**
https://bugzilla.mozilla.org/show_bug.cgi?id=1220978

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e697e0345a6381f7b4df8256b2a6e4ee716102b

File: apps/bluetooth/js/modules/pair_manager.js [hunk 1]
```diff
@@ -294,21 +295,24 @@ define(function(require) {
         showPairviewCallback: this.showPairview.bind(this, pairingInfo)
       };
       // Prepare notification toast.
-      var title = _('bluetooth-pairing-request-now-title');
-      var body = pairingInfo.evt.deviceName || _('unnamed-device');
+      var titleL10n = 'bluetooth-pairing-request-now-title';
+      var bodyL10n = pairingInfo.evt.deviceName ? 
+          { raw: pairingInfo.evt.deviceName } :
+          'unnamed-device';
```
Comment (zbraniecki) [inline apps/bluetooth/js/modules/pair_manager.js]: this file seems to be using 2-space indentation. Please, follow the files indentation patterns.

File: apps/bluetooth/js/modules/pair_manager_v1.js [hunk 2]
```diff
@@ -98,21 +100,24 @@ define(function(require) {
         showPairviewCallback: this.showPairview.bind(this, pairingInfo)
       };
       // Prepare notification toast.
-      var title = _('bluetooth-pairing-request-now-title');
-      var body = pairingInfo.name || _('unnamed-device');
+      var titleL10n = 'bluetooth-pairing-request-now-title';
+      var bodyL10n = pairingInfo.name ?
+          { raw: pairingInfo.name } :
+          'unnamed-device';
```
Comment (zbraniecki) [inline apps/bluetooth/js/modules/pair_manager_v1.js]: same here

File: apps/bluetooth/test/unit/modules/pair_manager_test.js [hunk 3]
```diff
@@ -357,12 +363,13 @@ suite('Bluetooth app > PairManager ', function() {
     var notificationStub, mockNotification, pairingInfo, titleResult,
       optionsResult;
     setup(function() {
-      var _ = window.navigator.mozL10n.get;
+//      var _ = window.navigator.mozL10n.get;
```
Comment (zbraniecki) [inline apps/bluetooth/test/unit/modules/pair_manager_test.js]: please, remove this line

**Comment by zbraniecki** [comment]
You should be able to remove `var _ = mozL10n.get` line from `apps/bluetooth/js/modules/pair_manager_v1.js` as well.

**Comment by hkasemir** [comment]
In line 211, the function `showPairView` uses _, so I don't think I can remove it

**Comment by zbraniecki** [comment]
ok, makes sense.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dda7cc698b62825c3e061fb6e40c14f03eabc4fb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b379c85ac406206bcbff4739b48a30f6b926dac5

File: apps/bluetooth/js/modules/pair_manager.js [hunk 4]
```diff
@@ -294,21 +295,24 @@ define(function(require) {
         showPairviewCallback: this.showPairview.bind(this, pairingInfo)
       };
       // Prepare notification toast.
-      var title = _('bluetooth-pairing-request-now-title');
-      var body = pairingInfo.evt.deviceName || _('unnamed-device');
+      var titleL10n = 'bluetooth-pairing-request-now-title';
+      var bodyL10n = pairingInfo.evt.deviceName ?
```
Comment (gasolin) [inline apps/bluetooth/js/modules/pair_manager.js]: trailing space

File: apps/bluetooth/js/modules/pair_manager_v1.js [hunk 5]
```diff
@@ -98,21 +100,24 @@ define(function(require) {
...
       var notificationId = 'pairing-request';
-      var notification = new Notification(title, {
-        body: body,
+      return NotificationHelper.send(titleL10n, {
+        bodyL10n: bodyL10n,
         icon: iconUrl,
         tag: notificationId
+      }).then(notification => {
+        // set onclick handler for the notification
+        notification.onclick =
```
Comment (gasolin) [inline apps/bluetooth/js/modules/pair_manager_v1.js]: trailing space

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f70d104473eb4731d92bf22658ccb6380ac50f5c

**Pull request merged**